### PR TITLE
Converted bug descriptions to YAML format (fixes #99)

### LINKED ISSUE #99: Update bug description files to use YAML file format

### PR DESCRIPTION
**Pull request by ChrisTimperley**
Fixes #99

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
I hope you used some automation for this :)

that's quite a few lines.

**Comment by gavanderhoorn** [comment]
Haven't tested it yet, but nice progress.

One nitpick: can we please not use submodules?

**Comment by ChrisTimperley** [comment]
> Haven't tested it yet, but nice progress.
> 
> One nitpick: can we please not use submodules?

I'm not a big fan of submodules either, but in this instance, it does make it easier to call out to the time machine (since we know where the script is located). I'd like to avoid the need to specify the location of the time machine script when calling `build-rosinstall.py`. We could copy the current version of the time machine to the location of its submodule, but that's not a very nice solution, either. Alternatively, we could assume that the time machine script can be found on the `PATH`? Any thoughts?

**Comment by gavanderhoorn** [comment]
We could create an install script that places the script in `/usr/loca/bin`. That would be immediately on the path.

Alternatively, we could place it in `$HOME/bin`. That is also on the `PATH` (on Ubuntu at least).

The install script would be a simple:

```bash
mkdir -p $HOME/bin
cp rosinstall_generator_tm.sh $HOME/bin
```

I think I'd prefer that over a submodule.

**Comment by ChrisTimperley** [comment]
That works. I'll update `build-rosinstall.py` to assume the time machine is on the PATH.

File: mavros/6bb2672.bug [hunk 1]
```diff
@@ -45,3 +45,8 @@ fix:
   languages:
     - C++
   time: 2017-03-30 (00:00)
+time-machine:
+  ros_distro: kinetic
+  ros_pkgs:
+    - mavros
+  datetime: 2017-03-30
```
Comment (gavanderhoorn) [inline mavros/6bb2672.bug]: Is this precise enough?

`rosdistro` receives a lot of commits on a single day.

If you use the [get_issue_creation_date.py](https://github.com/rosin-project/rosinstall_generator_time_machine/blob/245fcd6525e8dc302273885041ac9921b1352474/get_issue_creation_date.py) script, it should give you a full timestamp which would be more precise.

Comment (ChrisTimperley) [reply mavros/6bb2672.bug]: Doesn't `get_issue_creation_date.py` assume that there's an issue for the bug? A number of the mavros bugs don't have an issue.

Comment (ChrisTimperley) [reply mavros/6bb2672.bug]: Also, I'm having a few issues providing more specific ISO 8601 timestamps. When I pass `2015-12-04T02:20:03+03`, I get the following error:
```
executing command: rosinstall_generator_tm.sh 2015-12-03 23:20:03 c6791f0 indigo mavros /home/chris/bugs/robust/mavros/c6791f0/deps.rosinstall
Already on 'master'
Your branch is up-to-date with 'origin/master'.
Going back to '2015-12-03 23:20:03'
Provided date '2015-12-03 23:20:03' is not a valid date ..
````

Comment (gavanderhoorn) [reply mavros/6bb2672.bug]: > 
> 
> Doesn't `get_issue_creation_date.py` assume that there's an issue for the bug? A number of the mavros bugs don't have an issue.

No, it actually works for every resource on github that has a "created_at" property, which includes PRs as well. If it doesn't work, please report that.

Comment (gavanderhoorn) [reply mavros/6bb2672.bug]: > 
> 
> Also, I'm having a few issues providing more specific ISO 8601 timestamps. When I pass `2015-12-04T02:20:03+03`, I get the following error:
> 
> ```
> executing command: rosinstall_generator_tm.sh 2015-12-03 23:20:03 c6791f0 indigo mavros /home/chris/bugs/robust/mavros/c6791f0/deps.rosinstall
> Already on 'master'
> Your branch is up-to-date with 'origin/master'.
> Going back to '2015-12-03 23:20:03'
> Provided date '2015-12-03 23:20:03' is not a valid date ..
> ```

I'm a bit confused by the command line you show: there appears to be a space between the date and the time, so it effectively becomes two args. That won't work.

It also doesn't include timezone info, perhaps that is what trips up `date`

Comment (ChrisTimperley) [reply mavros/6bb2672.bug]: Yeah, that is strange. The time machine appears to be transforming the timestamp to UTC. I'll manually convert the stamps to UTC and see what happens.

Comment (gavanderhoorn) [reply mavros/6bb2672.bug]: But what's up with the whitespace in the command line?

Comment (ChrisTimperley) [reply mavros/6bb2672.bug]: Ah, my bad. I see the problem. The datetime is loaded as a `datetime` object in Python, which is pretty printed when it's added to the command string. I forgot that YAML would parse the datetime. I'll add a step to format the datetime correctly.

**Comment by ChrisTimperley** [comment]
> Would it be an idea to add the `.rosinstall` file in a separate PR?
> 
> This PR would then be only about converting the bug descriptions to YAML. The next one would 'use' those updated descriptions to generate those `.rosinstall` files.

Yeah, that's a good suggestion. This PR is getting quite messy.

**Pull request merged**